src/modules/powerrename/PowerRenameUILib/MainWindow.xaml.cpp

```diff
@@ -738,7 +740,8 @@ namespace winrt::PowerRenameUI::implementation
                 int id = 0;
```

Comment:
even better, instead of iterating here over all elements (GetItemCount), you can iterate over m_explorerItemsMap (or m_explorerItems), and check/unchech those items, as these collections are updated on ShowAll/ShowRenamed.

Reply:
The current ToggleAll function changes the PowerRenameItem selected and the ExplorerItem selected so I thought it would make it more complex to run the loop twice.

Reply:
not running the loop twice. Just loop through m_explorerItemsMap and then you can get PowerRenameItem with PRManger->GetById and change it that way

Reply:
It also needs to go through the ones in m_prManager that are not in m_explorerItemsMap to change its select state.

Reply:
There are 2 options here when showing only the files that will be renamed and clicking Select All/Unselect All:
 - Select All/Unselect All for all items that are loaded into PowerRename
 - Select All/Unselect All only the items that can be renamed
 
 Current behavior is #1 I think, right? But I'd prefer to go with #2, as it makes more sense from UX side. WDYT?

Reply:
Yes, it selects/unselects everything loaded in powerRename like in option 1.
`for (auto& [key, value] : m_explorerItemsMap)
        {
            value.Checked(selected);
        }`
https://user-images.githubusercontent.com/14083829/225715805-85d1c2a0-5415-467b-bc85-fa66acecb4c0.mp4


Option 2 may allow unchecking only what is being renamed, and selecting and unselecting it.
`for (UINT i = 0; i < itemCount; i++)
        {
            CComPtr<IPowerRenameItem> spItem;
            if (SUCCEEDED(m_prManager->GetItemByIndex(i, &spItem)))
            {
                spItem->PutSelected(selected);
                int id = 0;
                spItem->GetId(&id);
                auto item = FindById(id);
                if (item)
                    item.Checked(selected);
            }
        }`
https://user-images.githubusercontent.com/14083829/225715823-7fc157a5-bbc8-49fe-a4f2-ed7f27e1505e.mp4

But sometimes It feels not intuitive.
I still prefer option 1, check the vídeos.